dpytools/slack/slack.py

```diff
@@ -1,14 +1,25 @@
+import os
+import logging
+from dpytools.http_clients.base import BaseHttpClient
 
 class SlackNotifier:
 
     def __init__(self):
-        # Set a webhok via an env var, ask Mike for a 
-        #web hook url.
-        ...
-
-    def notify(self, msg: str):
-        # Check formatting options for messages to slack.
-        # From memory you style it via sending a dictionary.
-        # It's a post request so do use the http client
-        # we've developing elsewhere in this library.
-        ...
\ No newline at end of file
+        self.webhook_url = os.getenv("SLACK_WEBHOOK_URL")
+        if not self.webhook_url:
+            raise ValueError('SLACK_WEBHOOK_URL is not set')
+        self.http_client = BaseHttpClient()
+        self.validate_webhook_url()
+
+    def validate_webhook_url(self):
```

Comment:
just remove the need to do this please if we cant sent an OPTIONS or HEAD request. for now we can assume people are giving us sensible urls.